codex-ui/src/vue/components/alert/useAlert.ts

```diff
@@ -0,0 +1,35 @@
+import { ref } from 'vue';
+import { createSharedComposable } from '@vueuse/core';
+import type { AlertInterface, AlertOptions, Alertype } from './Alert.types';
+
+/**
+ * Alert service hook
+ */
+export const useAlert = createSharedComposable((): AlertInterface => {
+  const alerts = ref<AlertOptions[]>([]);
+
+  /**
+   * Trigger alert component
+   * @param type type of alert (success, error, warning, info and default)
+   * @param opt alert options(timeout, message and icon)
+   */
+  function triggerAlert(type?: Alertype, opt?: Pick<AlertOptions, 'icon' | 'message' | 'timeout'>): void {
+    const alertIndex = alerts.value.findIndex((idx: AlertOptions) => idx.timeout === opt?.timeout);
+
+    alerts.value.push({ type,
+      ...opt });
+
+    setTimeout(() => {
+      alerts.value.splice(alertIndex, alerts.value.shift() as number);
```

Comment:
`shift()` is redundant here, you can simplify it to improve readability

```suggestion
      alerts.value.splice(alertIndex, 1);
```